src/Deprecated90/BlockClosure.extension.st

```diff
@@ -0,0 +1,21 @@
+Extension { #name : #BlockClosure }
+
+{ #category : #'*Deprecated90' }
+BlockClosure >> ifError: errorHandlerBlock [
+
+	errorHandlerBlock numArgs = 0 
+		ifTrue: [ self deprecated: 'Use #onErrorDo: instead.' transformWith: '`@receiver ifError: `@arg' -> '`@receiver onErrorDo: `@arg' ]
+		ifFalse: [ self deprecated: 'Use #onErrorDo: instead.' ].
+		
+	^ self onErrorDo: errorHandlerBlock 
```

Comment:
You need to wrap it like:
```Smalltalk
self onErrorDo: [:err | errorHandlerBlock cull: err receiver cull: err description]
```
But I did not check in code what exact #cull:cull: it should be to be compatible.

Reply:
Why wrap? We just dispatch further on. Do you have a problematic case?

Reply:
That is original method:
```Smalltalk
ifError: errorHandlerBlock
   ^ self on: Error do: [:ex |
		errorHandlerBlock cull: ex description cull: ex receiver]
```
The issue is about eliminating this strange error unboxing so that the handler block will accept a normal error instance.

Reply:
what I am trying to say is that deprecated method should continue work with this strange two parameters

Reply:
Yes, the deprecated method should continue working as before :)